### ISSUE TITLE
Low epoch and minibatch accuracy

### ISSUE THREAD

**Issue by Fahimtonmoy** [ContinualAI/avalanche#1342]
After preprocess, below is my list of datasets (numpy array) which I have converted to torch tensors and created a benchmark.
 
While training the model, accuracy for some of the experiences are quite low and also epoch accuracy is not changing whereas, minibatch accuracy is fluctuating. I am stuck and not being able to identify my mistakes and what to change in the code. Could anyone kindly help. Thanks in advance.

```
datasets = [(x_goldeneye, y_goldeneye), (x_icmpflood, y_icmpflood), (x_ssh, y_ssh), (x_synflood, y_synflood), (x_synscan, y_synscan), (x_slowloris, y_slowloris), (x_tcpconnect, y_tcpconnect), (x_torshammer, y_torshammer), (x_udpflood, y_udpflood), (x_udpscan, y_udpscan)]
```
```
# Converting numpy arrays to tensors
train_input_tensors = []
train_target_tensors = []
task_labels = []
for i, (X, y) in enumerate(datasets):
    # Convert numpy arrays to PyTorch tensors
    X_tensor = torch.tensor(X, dtype=torch.float32)
    y_tensor = torch.tensor(y, dtype=torch.float32)  # Change dtype to torch.float32
    train_input_tensors.append(X_tensor)
    train_target_tensors.append(y_tensor)
    task_labels.append(i)
```
```
from avalanche.benchmarks.generators import tensors_benchmark
train_tensors = list(zip(train_input_tensors, train_target_tensors))
test_tensors = list(zip(train_input_tensors, train_target_tensors)) # Using the same data for train and test
created_benchmark = tensors_benchmark(
    train_tensors=train_tensors,
    test_tensors=test_tensors,
    task_labels=task_labels,
    complete_test_set_only=False
)
```
```
from avalanche.training.plugins import EvaluationPlugin
from avalanche.evaluation.metrics import accuracy_metrics
from avalanche.logging import InteractiveLogger
eval_plugin = EvaluationPlugin(
    accuracy_metrics(minibatch=True, epoch=True, experience=True, stream=True),
    loggers=[InteractiveLogger()]
)
```
```
def custom_bce_loss(input, target):
    target = target.reshape(-1, 1)
    target = target.float()
    return torch.nn.functional.binary_cross_entropy(input, target) 
from avalanche.training import Naive
optimizer = optim.Adam(model.parameters(), lr=0.01)
strategy = Naive(model=model, optimizer=optimizer, criterion= custom_bce_loss, train_mb_size=128, train_epochs=5, evaluator= eval_plugin)
```
```
for experience in created_benchmark.train_stream:
    print(f"Training on experience: {experience.current_experience}")
    strategy.train(experience) 
```
```
Training on experience: 0
-- >> Start of training phase << --
100%|██████████| 733/733 [00:07<00:00, 95.26it/s]
Epoch 0 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task000 = 0.1885
	Top1_Acc_MB/train_phase/train_stream/Task000 = 0.2710
100%|██████████| 733/733 [00:07<00:00, 92.61it/s] 
Epoch 1 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task000 = 0.1885
	Top1_Acc_MB/train_phase/train_stream/Task000 = 0.1495
100%|██████████| 733/733 [00:07<00:00, 98.26it/s]
Epoch 2 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task000 = 0.1885
	Top1_Acc_MB/train_phase/train_stream/Task000 = 0.2617
100%|██████████| 733/733 [00:08<00:00, 88.04it/s]
Epoch 3 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task000 = 0.1885
	Top1_Acc_MB/train_phase/train_stream/Task000 = 0.1776
100%|██████████| 733/733 [00:07<00:00, 97.50it/s]
Epoch 4 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task000 = 0.1885
	Top1_Acc_MB/train_phase/train_stream/Task000 = 0.2056
-- >> End of training phase << --
Training on experience: 1
-- >> Start of training phase << --
100%|██████████| 143/143 [00:01<00:00, 89.40it/s]
Epoch 0 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task001 = 0.9665
	Top1_Acc_MB/train_phase/train_stream/Task001 = 0.9515
100%|██████████| 143/143 [00:01<00:00, 95.07it/s] 
Epoch 1 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task001 = 0.9665
	Top1_Acc_MB/train_phase/train_stream/Task001 = 1.0000
100%|██████████| 143/143 [00:01<00:00, 104.58it/s]
Epoch 2 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task001 = 0.9665
	Top1_Acc_MB/train_phase/train_stream/Task001 = 0.9903
100%|██████████| 143/143 [00:01<00:00, 110.05it/s]
Epoch 3 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task001 = 0.9665
	Top1_Acc_MB/train_phase/train_stream/Task001 = 0.9612
100%|██████████| 143/143 [00:01<00:00, 102.96it/s]
Epoch 4 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task001 = 0.9665
	Top1_Acc_MB/train_phase/train_stream/Task001 = 0.9806
-- >> End of training phase << --
Training on experience: 2
-- >> Start of training phase << --
100%|██████████| 13/13 [00:00<00:00, 116.63it/s]
Epoch 0 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task002 = 1.0000
	Top1_Acc_MB/train_phase/train_stream/Task002 = 1.0000
100%|██████████| 13/13 [00:00<00:00, 112.97it/s]
Epoch 1 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task002 = 1.0000
	Top1_Acc_MB/train_phase/train_stream/Task002 = 1.0000
100%|██████████| 13/13 [00:00<00:00, 123.34it/s]
Epoch 2 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task002 = 1.0000
	Top1_Acc_MB/train_phase/train_stream/Task002 = 1.0000
100%|██████████| 13/13 [00:00<00:00, 113.07it/s]
Epoch 3 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task002 = 1.0000
	Top1_Acc_MB/train_phase/train_stream/Task002 = 1.0000
100%|██████████| 13/13 [00:00<00:00, 96.75it/s]
Epoch 4 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task002 = 1.0000
	Top1_Acc_MB/train_phase/train_stream/Task002 = 1.0000
-- >> End of training phase << --
Training on experience: 3
-- >> Start of training phase << --
100%|██████████| 349/349 [00:03<00:00, 106.55it/s]
Epoch 0 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task003 = 0.8926
	Top1_Acc_MB/train_phase/train_stream/Task003 = 0.8587
100%|██████████| 349/349 [00:03<00:00, 93.85it/s]
Epoch 1 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task003 = 0.8926
	Top1_Acc_MB/train_phase/train_stream/Task003 = 0.8913
100%|██████████| 349/349 [00:03<00:00, 91.06it/s] 
Epoch 2 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task003 = 0.8926
	Top1_Acc_MB/train_phase/train_stream/Task003 = 0.8913
100%|██████████| 349/349 [00:03<00:00, 103.18it/s]
Epoch 3 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task003 = 0.8926
	Top1_Acc_MB/train_phase/train_stream/Task003 = 0.8152
100%|██████████| 349/349 [00:03<00:00, 104.16it/s]
Epoch 4 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task003 = 0.8926
	Top1_Acc_MB/train_phase/train_stream/Task003 = 0.8913
-- >> End of training phase << --
Training on experience: 4
-- >> Start of training phase << --
100%|██████████| 90/90 [00:00<00:00, 107.81it/s]
Epoch 0 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task004 = 0.1250
	Top1_Acc_MB/train_phase/train_stream/Task004 = 0.0862
100%|██████████| 90/90 [00:00<00:00, 103.68it/s]
Epoch 1 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task004 = 0.1250
	Top1_Acc_MB/train_phase/train_stream/Task004 = 0.1207
100%|██████████| 90/90 [00:00<00:00, 92.90it/s]
Epoch 2 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task004 = 0.1250
	Top1_Acc_MB/train_phase/train_stream/Task004 = 0.2414
100%|██████████| 90/90 [00:00<00:00, 95.03it/s]
Epoch 3 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task004 = 0.1250
	Top1_Acc_MB/train_phase/train_stream/Task004 = 0.1724
100%|██████████| 90/90 [00:01<00:00, 84.28it/s]
Epoch 4 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task004 = 0.1250
	Top1_Acc_MB/train_phase/train_stream/Task004 = 0.0517
-- >> End of training phase << --
Training on experience: 5
-- >> Start of training phase << --
100%|██████████| 243/243 [00:02<00:00, 90.70it/s] 
Epoch 0 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task005 = 0.7999
	Top1_Acc_MB/train_phase/train_stream/Task005 = 0.7949
100%|██████████| 243/243 [00:02<00:00, 104.53it/s]
Epoch 1 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task005 = 0.7999
	Top1_Acc_MB/train_phase/train_stream/Task005 = 0.7436
100%|██████████| 243/243 [00:02<00:00, 102.62it/s]
Epoch 2 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task005 = 0.7999
	Top1_Acc_MB/train_phase/train_stream/Task005 = 0.8205
100%|██████████| 243/243 [00:02<00:00, 101.31it/s]
Epoch 3 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task005 = 0.7999
	Top1_Acc_MB/train_phase/train_stream/Task005 = 0.8205
100%|██████████| 243/243 [00:02<00:00, 101.36it/s]
Epoch 4 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task005 = 0.7999
	Top1_Acc_MB/train_phase/train_stream/Task005 = 0.8974
-- >> End of training phase << --
Training on experience: 6
-- >> Start of training phase << --
100%|██████████| 91/91 [00:00<00:00, 97.18it/s]
Epoch 0 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task006 = 0.1394
	Top1_Acc_MB/train_phase/train_stream/Task006 = 0.1760
100%|██████████| 91/91 [00:01<00:00, 85.07it/s]
Epoch 1 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task006 = 0.1394
	Top1_Acc_MB/train_phase/train_stream/Task006 = 0.1280
100%|██████████| 91/91 [00:01<00:00, 84.77it/s]
Epoch 2 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task006 = 0.1394
	Top1_Acc_MB/train_phase/train_stream/Task006 = 0.1600
100%|██████████| 91/91 [00:01<00:00, 85.41it/s]
Epoch 3 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task006 = 0.1394
	Top1_Acc_MB/train_phase/train_stream/Task006 = 0.0720
100%|██████████| 91/91 [00:01<00:00, 88.06it/s]
Epoch 4 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task006 = 0.1394
	Top1_Acc_MB/train_phase/train_stream/Task006 = 0.1360
-- >> End of training phase << --
Training on experience: 7
-- >> Start of training phase << --
100%|██████████| 298/298 [00:02<00:00, 101.99it/s]
Epoch 0 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task007 = 0.2160
	Top1_Acc_MB/train_phase/train_stream/Task007 = 0.1827
100%|██████████| 298/298 [00:02<00:00, 103.04it/s]
Epoch 1 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task007 = 0.2160
	Top1_Acc_MB/train_phase/train_stream/Task007 = 0.2115
100%|██████████| 298/298 [00:02<00:00, 100.47it/s]
Epoch 2 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task007 = 0.2160
	Top1_Acc_MB/train_phase/train_stream/Task007 = 0.1635
100%|██████████| 298/298 [00:03<00:00, 93.09it/s]
Epoch 3 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task007 = 0.2160
	Top1_Acc_MB/train_phase/train_stream/Task007 = 0.3173
100%|██████████| 298/298 [00:03<00:00, 84.47it/s]
Epoch 4 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task007 = 0.2160
	Top1_Acc_MB/train_phase/train_stream/Task007 = 0.1827
-- >> End of training phase << --
Training on experience: 8
-- >> Start of training phase << --
100%|██████████| 3655/3655 [00:40<00:00, 90.15it/s]
Epoch 0 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task008 = 0.6241
	Top1_Acc_MB/train_phase/train_stream/Task008 = 0.4000
100%|██████████| 3655/3655 [00:41<00:00, 88.70it/s]
Epoch 1 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task008 = 0.6241
	Top1_Acc_MB/train_phase/train_stream/Task008 = 0.2000
100%|██████████| 3655/3655 [00:40<00:00, 90.15it/s]
Epoch 2 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task008 = 0.6241
	Top1_Acc_MB/train_phase/train_stream/Task008 = 0.8000
100%|██████████| 3655/3655 [00:41<00:00, 87.82it/s]
Epoch 3 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task008 = 0.6241
	Top1_Acc_MB/train_phase/train_stream/Task008 = 0.6000
100%|██████████| 3655/3655 [00:40<00:00, 89.67it/s]
Epoch 4 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task008 = 0.6241
	Top1_Acc_MB/train_phase/train_stream/Task008 = 0.6000
-- >> End of training phase << --
Training on experience: 9
-- >> Start of training phase << --
100%|██████████| 79/79 [00:00<00:00, 103.00it/s]
Epoch 0 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task009 = 0.2147
	Top1_Acc_MB/train_phase/train_stream/Task009 = 0.1695
100%|██████████| 79/79 [00:00<00:00, 104.89it/s]
Epoch 1 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task009 = 0.2147
	Top1_Acc_MB/train_phase/train_stream/Task009 = 0.1525
100%|██████████| 79/79 [00:00<00:00, 103.86it/s]
Epoch 2 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task009 = 0.2147
	Top1_Acc_MB/train_phase/train_stream/Task009 = 0.1356
100%|██████████| 79/79 [00:00<00:00, 99.19it/s]
Epoch 3 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task009 = 0.2147
	Top1_Acc_MB/train_phase/train_stream/Task009 = 0.2034
100%|██████████| 79/79 [00:00<00:00, 98.63it/s] 
Epoch 4 ended.
	Top1_Acc_Epoch/train_phase/train_stream/Task009 = 0.2147
	Top1_Acc_MB/train_phase/train_stream/Task009 = 0.2373
-- >> End of training phase << --
```

**Comment by AntonioCarta** [comment]
Hey, I'm sorry but we use the issues only for Avalanche bugs. This is a model selection problem and there is not much that we can do about it.
